.changeset/many-weeks-sort.md

```diff
@@ -0,0 +1,43 @@
+---
+'astro': minor
+---
+
+Form support in View Transitions router
+
+The `<ViewTransitions />` router can now handle form submissions, allowing the same animated transitions and stateful UI retention you already get from `<a>` links to occur on form posts. With this added in Astro you can have animations in all of these scenarios:
+
+- Clicking links between pages.
+- Making stateful changes in forms (ie updating site preferences).
+- Manually triggered navigate via the `navigate()` API.
+
+This feature is opt-in for semver reasons and can be enabled by adding the `handleForms` prop to the component like so:
```

Comment:
```suggestion
The `<ViewTransitions />` router can now handle form submissions, allowing the same animated transitions and stateful UI retention on form posts that are already available on `<a>` links. With this addition, your Astro project can have animations in all of these scenarios:

- Clicking links between pages.
- Making stateful changes in forms (i.e. updating site preferences).
- Manually triggering navigation via the `navigate()` API.

This feature is opt-in for semver reasons and can be enabled by adding the `handleForms` prop to the `<ViewTransitions /> component:
```
Just suggested some tiny word changes. 

And just want to double check: e.g. should be used if the thing in brackets is one example ( of many possibilities), and i.e. is used when it's literally exactly the thing it is, just rephrased. 

So, if "making stateful changes **=** "updating site preferences", then i.e. is fine. Otherwise, that should be e.g.

Reply:
Ok, so this is one example, so I'll change to e.g. after merging.